### ISSUE TITLE
Rule: Credentials should not be hard-coded

### ISSUE THREAD

**Issue by alban-auzeill** [SonarSource/SonarJS#950]
Implements [RSPEC-2068](https://jira.sonarsource.com/browse/RSPEC-2068)

**Comment by eric-therond-sonarsource** [comment]
Should raises when a non **empty/ non trivial string** value is assigned to a relevant variable/property.

Examples:
```
var guestConnection = mysql.createConnection(
{
  host:'localhost',
  user: args.gu,
  database: args.d,
  password: "mypassword", // sensitive 
  multipleStatements: true
});
```
Should be able to detect the call to mysql.createConnection method with an argument (an object)  which contains a hardcoded password assigned to a defined property.

Should also be able to detect hardcoded passwords in json files like here:
```
{
    "auths": {
        "https://index.docker.io/v1/": {
            "email": "docker@example.com",
            "auth": "X3Rva2VuOjEyMzQuMThqZjg0MWZrbDQwYU90dTNrLXdCbDVuaThDM2Q0QVh0QjM2V2VqZzM4MDA2WlR5TDhUOWg5VXgrWWwzdTNVQ1hDWFZlWg__SENSITIVE__"
        },
        "https://hub.docker.com/": {
            "email": "docker@example.com",
            "auth": "X3Rva2VuOjEyMzQuMThqZjg0MWZrbDQwYU90dTNrLXdCbDVuaThDM2Q0QVh0QjM2V2VqZzM4MDA2WlR5TDhUOWg5VXgrWWwzdTNVQ1hDWFZlWg__SENSITIVE__"
        }
    }
} 
```

Do not raise when variable is empty or variable name equals variable value:
```
string password = ""; // Compliant
string pwd = ""; // Compliant
string password = "password"; // Compliant
string pwd = "pwd"; // Compliant
```

Code examples:
https://github.com/SonarSource/security-expected-issues/tree/master/javascript/hotspots/RSPEC-2068

**Comment by ilia-kebets-sonarsource** [comment]
we will drop hotspots, so we might consider dropping this issue.

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-141
